### PR TITLE
3dmerge

### PR DESCRIPTION
**Pull request by TalPat**
3d renderer with single enemy and bomb support. Basic placeholder animations in place.

### CODE REVIEW

**Review by WSeegers — COMMENTED** [review]
Minor issue on the includes folder. This folder should be files relating to the public api, so anything in the includes should be safe and meaningfully to users from outside the library to call and use.

**Comment by MEWestvig** [comment]
Just fix this warning. It is annoying me - Daniel

```
TalonBomberman/renderer3D/src/Animation.cpp:33:1: warning:
      control reaches end of non-void function [-Wreturn-type]
}
```

**Review by TalPat — COMMENTED** [review]
Includes have been rearranged, and value returned for disintegrate function. Unnecessary models and debugging lines removed

File: app/src/Bomberman.cpp [hunk 1]
```diff
@@ -8,10 +8,20 @@ const int MAP_WIDTH = 11;
 const int MAP_HEIGHT = 11;
 
 Bomberman::Bomberman()
-	: window(sf::VideoMode(WINDOW_WIDTH, WINDOW_HEIGHT), WINDOW_TITLE),
-	  renderTime(0),
+	: renderTime(0),
 	  engineTime(0)
 {
+	sf::ContextSettings settings;
+  settings.depthBits = 32;
```
Comment (GJSteyn) [inline app/src/Bomberman.cpp]: Inconsistent whitespace

File: app/src/Bomberman.hpp [hunk 2]
```diff
@@ -7,7 +7,7 @@
 #include <Engine.hpp>
 #include <EngineEvent.hpp>
 #include <Renderer.hpp>
-#include <SFML/Graphics.hpp>
+//#include <SFML/Graphics.hpp>
```
Comment (GJSteyn) [inline app/src/Bomberman.hpp]: Could we remove this comment?

File: app/src/Bomberman.hpp [hunk 3]
```diff
@@ -20,7 +20,7 @@ static const char *WINDOW_TITLE = "Bomberman";
 class Bomberman : private AMainLoop
 {
 private:
-	sf::RenderWindow window;
+	sf::RenderWindow* window;
```
Comment (GJSteyn) [inline app/src/Bomberman.hpp]: Don't mean to be pedantic, but this use of asterisk is inconsistent with other examples in PR.

File: renderer/include/Renderer.hpp [hunk 4]
```diff
@@ -1,20 +1,45 @@
-#ifndef _Renderer_hpp_
-#define _Renderer_hpp_
+#ifndef RENDERER_HPP
```
Comment (GJSteyn) [inline renderer/include/Renderer.hpp]: Any good reason for this change?

File: renderer/include/Renderer.hpp [hunk 5]
```diff
@@ -1,20 +1,45 @@
...
+};
 
 class Renderer
 {
-public:
-	void render(sf::RenderWindow &window, const GameState &state);
 private:
+  std::vector<Model_st> _models;
+  Shader* _shader;
+  Camera* _camera;
```
Comment (GJSteyn) [inline renderer/include/Renderer.hpp]: Couple examples of inconsistent whitespace in this file.

File: renderer/include/Renderer.hpp [hunk 6]
```diff
@@ -1,20 +1,45 @@
...
+public:
+  Renderer(/* args */);
+  ~Renderer();
+  void init();
+  void render(sf::RenderWindow &window, const GameState &state);
 };
 
-#endif
+
+#endif
```
Comment (GJSteyn) [inline renderer/include/Renderer.hpp]: Just for the sake of consistency, could we get a newline at the end of this file?

File: renderer/src/Animation.cpp [hunk 7]
```diff
@@ -0,0 +1,71 @@
+#include "Animation.hpp"
```
Comment (GJSteyn) [inline renderer/src/Animation.cpp:1]: This file has a few interesting quirks:
- Java style curly braces
- Two-space indentation
- Missing single space before only some curly braces before function bodies

File: renderer/src/Animation.hpp [hunk 8]
```diff
@@ -0,0 +1,37 @@
+#ifndef ANIMATION_HPP
```
Comment (GJSteyn) [inline renderer/src/Animation.hpp:1]: Couple inconsistencies with formatting in this file

File: renderer/src/Camera.cpp [hunk 9]
```diff
@@ -0,0 +1,64 @@
+#include "Camera.hpp"
```
Comment (GJSteyn) [inline renderer/src/Camera.cpp:1]: Similar issues as Animation.cpp

**Review by GJSteyn — COMMENTED** [review]
I've left a few comments, all of which are in regards to formatting and style - although there are more examples that I didn't bother commenting on.
I want to suggest we start using some sort of linter for the sake of consistency.

Otherwise, the code seems fine. Works well on my end. Quite a big diff, so considering this an initial implementation for 3D would be fine with me.

**Review by TalPat — COMMENTED** [review]
Formatting has been handled by the Microsoft extension for c++ using default settings.

**Review by WSeegers — APPROVED** [review]

**Review by MEWestvig — APPROVED** [review]

**Pull request merged**